src/SharpGLTF.Core/Schema2/agi.Articulations.cs

```diff
@@ -0,0 +1,140 @@
+        {
+            var articulation = new AgiArticulation();
+            articulation.Name = name;
+
+            _articulations.Add(articulation);
+
+            return articulation;
+        }
+    }
+
+    public sealed partial class AgiNodeArticulations
+    {
+        internal AgiNodeArticulations(Node node) { }
+
+        public String ArticulationName
+        {
+            get => _articulationName;
+            set => _articulationName = value;
+        }
+
+        public Boolean? IsAttachPoint
+        {
+            get => _isAttachPoint;
+            set => _isAttachPoint = value;
+        }
+    }
+
+    public sealed partial class AgiArticulation : IChildOf<AgiRootArticulations>
+    {
+        internal AgiArticulation()
+        {
+            _stages = new ChildrenCollection<AgiArticulationStage, AgiArticulation>(this);
+        }
+
+        protected override IEnumerable<ExtraProperties> GetLogicalChildren()
+        {
+            return base.GetLogicalChildren().Concat(_stages);
+        }
+
+        public IReadOnlyList<AgiArticulationStage> Stages => _stages;
+
+        public AgiArticulationStage CreateArticulationStage(string name)
+        {
+            var stage = new AgiArticulationStage();
+            stage.Name = name;
+
+            _stages.Add(stage);
+
+            return stage;
+        }
+
+        public String Name
+        {
+            get => _name;
+            set => _name = value;
+        }
+
+        public Vector3? PointingVector
+        {
+            get => _pointingVector;
+            set => _pointingVector = value;
+        }
+
+        public int LogicalIndex { get; private set; } = -1;
+
+        public AgiRootArticulations LogicalParent { get; private set; }
+
+        public void _SetLogicalParent(AgiRootArticulations parent, int index)
+        {
+            LogicalParent = parent;
+            LogicalIndex = index;
+        }
+    }
+
+    public sealed partial class AgiArticulationStage : IChildOf<AgiArticulation>
+    {
+        public int LogicalIndex { get; private set; } = -1;
+
+        public AgiArticulation LogicalParent { get; private set; }
+
+        public void _SetLogicalParent(AgiArticulation parent, int index)
+        {
+            LogicalParent = parent;
+            LogicalIndex = index;
+        }
+
+        public String Name
+        {
+            get => _name;
+            set => _name = value;
+        }
+
+        public AgiArticulationTransformType TransformType
+        {
+            get => _type;
+            set => _type = value;
+        }
+
+        public Double MinimumValue
+        {
+            get => _minimumValue;
+            set => _minimumValue = value;
+        }
+
+        public Double InitialValue
+        {
+            get => _initialValue;
+            set => _initialValue = value;
+        }
+
+        public Double MaximumValue
+        {
+            get => _maximumValue;
+            set => _maximumValue = value;
+        }
```

Comment:
The json parser I'm using works with doubles, but I usually try to expose the values as floats, unless there's some special precission requirement, I don't know if this is related.

Also, if there's a relationship between MinimumValue, InitialValue and MaximumValue, I use to remove the setters of the properties, and write a method for setting the three values in one shot, which allows cross checking the arguments, like checking that minimum is less than maximum and things like that

Reply:
I added the three-value setter with guards, in place of individual setters.

These values are for transformations, not vertex attributes, so they should remain as doubles.  They are read, written, and used as doubles, so there's no advantage to round-tripping them through single precision as one would do with vertex attributes and such.